lib/spack/spack/version.py

```diff
@@ -599,15 +599,26 @@ def satisfies(self, other):
         """A Version 'satisfies' another if it is at least as specific and has
         a common prefix.  e.g., we want gcc@4.7.3 to satisfy a request for
         gcc@4.7 so that when a user asks to build with gcc@4.7, we can find
-        a suitable compiler.
+        a suitable compiler. In the case of two GitVersions we require the ref_versions
+        to satisify one another and the versions to be an exact match.
         """
+
         self_cmp = self._cmp(other.ref_lookup)
         other_cmp = other._cmp(self.ref_lookup)
 
+        if other.is_ref:
+            # the version for a ref is the git ref, and it should be an exact match
+            # git hashes, and tags need to be exact matches
+            version_match = self.version == other.version
+        else:
+            # other is not a ref then it is a version base and we need to compare
+            # this.ref
+            version_match = self.ref_version == other.version
```

Comment:
```suggestion
        if self.is_ref and other.is_ref:
            # the version for a ref is the git ref, and it should be an exact match
            # git hashes, and tags need to be exact matches
            version_match = self.version == other.version
        elif self.is_ref:
            # other is not a ref then it is a version base and we need to compare
            # this.ref
            version_match = self.ref_version == other.version
        else:
            version_match = True
```
I think we have to watch out for situations in which `self.is_ref == False` because the version already got converted to `GitVersion` for some earlier comparison

Reply:
@becker33 I adjusted your changes a little and improved the comment.